src/coreclr/gc/gcpriv.h

```diff
@@ -5961,3 +5961,15 @@ class card_marking_enumerator
 #else
 #define THIS_ARG
 #endif // FEATURE_CARD_MARKING_STEALING
+
+template <typename T, typename U>
+auto max(T&& t, U&& u) -> decltype(t > u ? t : u)
+{
+    return t > u ? t : u;
+}
+
+template <typename T, typename U>
+auto min(T&& t, U&& u) -> decltype(t < u ? t : u)
+{
+    return t < u ? t : u;
+}
```

Comment:
There is a lot of subtlty going on with this code. What is the cost of not doing this? I am concerned this is going to introduce subtle bugs.

Reply:
These min/max overloads allow us to use (same-signedness) integer promotion rules in our min/max usage, which is the vast majority of the usage in the GC and the JIT (the other place I have a copy of these). Differing-signedness promotion is detected and produces an error in our GCC build leg (and the Win-x86 build leg also for some reason). These locations relied on the macro-based prior implementation to allow all of the integer promotion rules.

I could move every single min/max call to have exact type matches with manual casts for temporaries, but the number of places would quickly become a large portion of the changed lines in this PR and would make the code more difficult to read.

Reply:
I'm taking a look at removing these now. It's not as bad now that I've already had to adjust for signedness, so it's more just tedious. It's doable though, so I'm going through it.